src/Lexer.ts

```diff
@@ -0,0 +1,98 @@
+// @author Anush kumar N <anush.kumar@codingmart.com>
+
+import TokenType from './Token';
+
+type arrayOfString = Array<string>;
+type literalWithType = { type: Symbol; literal: string | undefined };
+type identifyLiteralReturnType = { token: literalWithType; updatedPos: number | undefined };
+
+// Performs lexical analysis on raw string input.
+function* Lexer(input: string) {
+  const inputAsArr: arrayOfString = Array.from(input);
+  inputAsArr.push(TokenType.eof.toString());
+
+  let position: number = 0;
```

Comment:
Why is `position` outside the for loop?

Reply:
oops it was by mistake! let me fix it.

Reply:
The curious thing is that the unit tests still passed with this "mistake"...

I don't have much experience using fn generators, but is it doing something that makes this work?

Reply:
Actually, it works the same with or without the mistake. The generator is executed only once. When ever the yield is called, Function will be paused  and will yield the result. The scope of the function is preserved will also be preserved until the generator is completely terminated.